nwb_conversion_tools/tools/neo/neo.py

```diff
@@ -0,0 +1,608 @@
+
+    if getattr(nwbfile, "icephys_sequential_recordings", None):
+        sessions_offset = len(nwbfile.icephys_sequential_recordings)
+    else:
+        sessions_offset = 0
+
+    relative_session_start_time = metadata["Icephys"]["Sessions"][sessions_offset]["relative_session_start_time"]
+    session_stimulus_type = metadata["Icephys"]["Sessions"][sessions_offset]["stimulus_type"]
+
+    # Sequential icephys recordings
+    simultaneous_recordings = list()
+    for si in range(n_segments):
+        # Parallel icephys recordings
+        recordings = list()
+        for ei, electrode in enumerate(
+            list(nwbfile.icephys_electrodes.values())[: len(neo_reader.header["signal_channels"]["units"])]
+        ):
+            # Starting time is the signal starting time within .abf file + time
+            # relative to first session (first .abf file)
+            ri += 1
+            starting_time = neo_reader.get_signal_t_start(block_index=0, seg_index=si)
+            starting_time = starting_time + relative_session_start_time
+
+            sampling_rate = neo_reader.get_signal_sampling_rate()
+            response_unit = neo_reader.header["signal_channels"]["units"][ei]
+            response_conversion = get_conversion_from_unit(unit=response_unit)
+            response_gain = neo_reader.header["signal_channels"]["gain"][ei]
+            response_name = f"{icephys_experiment_type}-response-{si + 1 + simultaneous_recordings_offset}-ch-{ei}"
+
+            response = response_classes[icephys_experiment_type](
+                name=response_name,
+                description=f"Response to: {session_stimulus_type}",
+                electrode=electrode,
+                data=neo_reader.get_analogsignal_chunk(block_index=0, seg_index=si, channel_indexes=ei),
+                starting_time=starting_time,
+                rate=sampling_rate,
+                conversion=response_conversion * response_gain,
+                gain=1.0,
+            )
+            if icephys_experiment_type != "izero":
+                stim_unit = protocol[2][ei]
+                stim_conversion = get_conversion_from_unit(unit=stim_unit)
+                stimulus = stim_classes[icephys_experiment_type](
+                    name=f"stimulus-{si + 1 + simultaneous_recordings_offset}-ch-{ei}",
+                    description=f"Stim type: {session_stimulus_type}",
+                    electrode=electrode,
+                    data=protocol[0][si][ei],
+                    rate=sampling_rate,
+                    starting_time=starting_time,
+                    conversion=stim_conversion,
+                    gain=1.0,
+                )
+                icephys_recording = nwbfile.add_intracellular_recording(
+                    electrode=electrode, response=response, stimulus=stimulus
+                )
+            else:
+                icephys_recording = nwbfile.add_intracellular_recording(electrode=electrode, response=response)
```

Comment:
Could be a tad nicer to only have a single call to `add_intracellular_recording` that takes `**icephys_recording_kwargs` where `icephys_recording_kwargs` is a `dict` that gets updated if included a `stimulus`, but always contains `response`.

Reply:
Doing so could also allow easy merging with any metadata that was manually specifies for the series; `icephys_recording_kwargs  = deep_dict_update(icephys_recording_kwargs , metadata["Icephys"]["PatchClampSeries"][si])` for instance

Not that this is required in this first PR, we can add it in later, but it would nice to start off with the base code structure in an easily modifiable way.